File: Day 1/6/calculate_factorial.c
#include <stdio.h>

int calculateFactorial(int n) {
    int factorial = 1;
    for (int i = 1; i <= n; i++) {
        factorial *= i;
    }
    return factorial;
}

int main() {
    int n;

    printf("Enter a non-negative integer (0 ≤ n ≤ 10): ");
    scanf("%d", &n);

    if (n < 0 || n > 10) {
        printf("Invalid input. Please enter a number between 0 and 10.\n");
    } else {
        printf("Factorial of %d: %d\n", n, calculateFactorial(n));
    }

    return 0;
}

File: Day 1/6/calculateFactorial.js
const calculateFactorial = (n) => {
    if (n === 0) return 1;
    return Array.from({ length: n }, (_, i) => i + 1)
                .reduce((product, num) => product * num, 1);
}

const calculateFactorialAlgorithmically = (n) => {
    let factorial = 1;
    for (let i = 1; i <= n; i++) {
        factorial *= i;
    }
    return factorial;
}

console.log(calculateFactorial(5));
console.log(calculateFactorialAlgorithmically(5));